clients/python/src/model_registry/models/base_resource_list.py

```diff
@@ -0,0 +1,65 @@
+from __future__ import annotations
+from dataclasses import dataclass, field
+from kiota_abstractions.serialization import Parsable, ParseNode, SerializationWriter
+from typing import Any, Callable, Dict, List, Optional, TYPE_CHECKING, Union
+
+if TYPE_CHECKING:
+    from .registered_model_list import RegisteredModelList
+
+from .registered_model_list import RegisteredModelList
+
+@dataclass
+class BaseResourceList(RegisteredModelList):
```

Comment:
IIUC the erratic behaviour on this generation might stem from it inheriting the items object (and not the other way around). This seems particularly wrong when comparing to the other models.